go-controller/pkg/node/node_ip_handler.go

```diff
@@ -0,0 +1,118 @@
+package node
+
+import (
+	utilnet "k8s.io/utils/net"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+type addressManager struct {
+	addresses      sets.String
+	nodeAnnotator  kube.Annotator
+	mgmtPortConfig *managementPortConfig
+}
+
+// initializes a new address manager which will hold all the IPs on a node
+func newAddressManager(nodeAnnotator kube.Annotator, config *managementPortConfig) (*addressManager, error) {
+	mgr := &addressManager{
+		addresses:      sets.NewString(),
+		nodeAnnotator:  nodeAnnotator,
+		mgmtPortConfig: config,
+	}
+	addrs, err := netlink.AddrList(nil, netlink.FAMILY_ALL)
```

Comment:
Hmmm. So there is a potential for you to miss updates between calling `newAddressManager` and `Run()`. You'd have to be rather unlucky to hit that though.

To be safe, you could move this init logic to after the `AddrSubscribe`

Reply:
good point. I'll do that